### DISCUSSION TITLE
Conveniently provide active build types to port files

### DISCUSSION THREAD

**Discussion by dg0yt** [microsoft/vcpkg#25384] [Ideas]
**Is your feature request related to a problem? Please describe.** 
To deal with the actually selected build types, port files and maintainer functions need to write:
~~~cmake
if(NOT DEFINED VCPKG_BUILD_TYPE OR VCPKG_BUILD_TYPE STREQUAL "debug")
...
endif()
if(NOT DEFINED VCPKG_BUILD_TYPE OR VCPKG_BUILD_TYPE STREQUAL "release")
...
endif()
~~~

This is rather long for a common condition and easy to get wrong (`DEFINED`? `Release`?).
This is one of the barriers for properly supporting single-config triplets in portfiles.

**Proposed solution**
Provide a list variable with the active build types.
~~~cmake
if("debug" IN_LIST VCPKG_ACTIVE_BUILD_TYPES)
...
endif()
if("release" IN_LIST VCPKG_ACTIVE_BUILD_TYPES)
...
endif()
~~~

**Describe alternatives you've considered**
`VCPKG_BUILD_TYPES` would be shorter but invites typos due to similarity with VCPKG_BUILD_TYPES. 

Discrete boolean variables would allow even shorter conditions, and remove ambiguity about spelling/case:
~~~cmake
if(VCPKG_BUILD_DEBUG)
...
endif()
if(VCPKG_BUILD_RELEASE)
...
endif()
~~~

**Comment by autoantwort** [comment]
I strongly prefer `VCPKG_BUILD_DEBUG` and `VCPKG_BUILD_RELEASE` :)

**Comment by Neumann-A** [comment]
related  #7733

**Comment by dg0yt** [comment]
> related #7733

That PR includes per-build-type properties which I also considered but didn't want to add to this specific issue. :+1:
